decisions/0005-remixing-react-router.md

```diff
@@ -0,0 +1,202 @@
+
+**Backwards Compatibility**
+
+This will be handled in the deprecated `useTransition` hook along with the backfill of `type` and `submission` properties
+
+### Form automatic replace behavior
+
+When performing POST navigations, you don't want to end up with a duplicate entry in the history stack which makes back-button routing weird when going through the same page twice. This is further complicated in browsers that hang onto the submission info and thus have to prompt you to warn you of re-submitting your data. We'll look at a few examples to demonstrate, but the intention is that the default behavior of the router should ensure that you can't get yourself into this double-history-entry situation when using `<Form method="post">` submission navigations.
+
+Normal POST submissions that do not redirect will use a `REPLACE`:
+
+- User is on `/` (history stack is `[/]`)
+- Navigates to `/login` (history stack is `[/, /login]`)
+- Fills out and submits the `<Form method="post">`
+- Action does not redirect
+  - At this point, if the action returns a non-redirect and we were to PUSH the navigation we'd end up with a history stack of `[/, /login, /login]` and the user would be in a scenario where it would take them 2 back buttons to get "through" the login page from a subsequent route.
+  - To avoid this, when a POST submission does not return a redirect, the router will REPLACE in the history stack, leaving us at `[/, /login]` and avoiding the duplicate history entry
+
+Normal POST submissions that _do_ redirect will use `PUSH` for the redirect:
+
+- User is on `/` (history stack is `[/]`)
+- Navigates to `/login` (history stack is `[/, /login]`)
+- Fills out and submits the `<Form method="post">`
+- Action redirects to `/private`
+  - If we treated this redirect as a REPLACE, we'd be replacing the _initial_ navigation to `/login` since we haven't yet touched history for the POST. This would leave the history stack as `[/, /private]` and we'd lose the fact that we were ever at the login page.
+  - Instead when an action redirects, we'll use a PUSH and in this case the history stack would become `[/, /login, /private]` and the user would be able to navigate back through the login page and to the home page
+
+Note: User's can still be explicit here and use `<Form method="post" replace={shouldReplace}>` and the router will respect the value passed to `replace`.
+
+### `unstable_shouldReload` stabilized as `shouldRevalidate`
+
+We stabilized the API for when a given route loader should re-run, and changed the name to align with the "revalidation" nomenclature and the `useRevalidator` hook. We also leave more control in the hands of the user here. In Remix there were some cases in which you _could not_ opt out of revalidation and if your method did run you had full control and couldn't necessarily handle one edge case and then say "do what you otherwise would have done".
```

Comment:
```suggestion
We stabilized the API for when a given route loader should re-run, and changed the name to align with the "revalidation" nomenclature and the `useRevalidator` hook. We also leave more control in the hands of the user here. In Remix there were some cases in which you _could not_ opt out of revalidation and if your method did run, you had full control and couldn't necessarily handle one edge case and then say "do what you otherwise would have done".
```